### ISSUE TITLE
Occasional, continuous major GC causing the the cluster unresponsive

### ISSUE THREAD

**Issue by jongwook** [hazelcast/hazelcast#5906]
I have a 5-node Hazelcast 3.4.3 cluster which serves heavy QueryProcessor loads, and I'm confronting an issue that the entire cluster dies very occasionally. (once per 2~3 weeks)

The graph below, showing each node's gc counts, gc times and the cluster's query throughput, indicates that one node went unresponsive, and the other four went constantly gc-ing 10 minutes later, presumably because of the operation queue size waiting on the first node.

![graph](https://cloud.githubusercontent.com/assets/266841/9164647/bfd6f592-3f88-11e5-80ad-5f369ca170e0.png)

The command line options were:

```
/usr/bin/java -server -XX:+UseG1GC -XX:G1HeapWastePercent=5 -XX:ConcGCThreads=8 -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.port=4599 -Xms8g -Xmx8g com.hazelcast.core.server.StartServer
```

Does anyone have any clues or solution to this problem?

**Comment by bwzhang2011** [comment]
@mesutcelik, hazelcast team should provide the use some good choice for jvm adjusment towards different jdk between jdk1.6  to jdk1.8.

**Comment by bwzhang2011** [comment]
@mesutcelik, any update with such issue ?

**Comment by bwzhang2011** [comment]
@jongwook, any update with such issue ?

**Comment by mdogan** [comment]
You can enable diagnostics to be able to see internal details of the system during failure.
```
-Dhazelcast.diagnostics.enabled=true
-Dhazelcast.diagnostics.metric.level=info
-Dhazelcast.diagnostics.invocation.sample.period.seconds=30
-Dhazelcast.diagnostics.pending.invocations.period.seconds=30
-Dhazelcast.diagnostics.slowoperations.period.seconds=30
```

See http://docs.hazelcast.org/docs/3.8.2/manual/html-single/index.html#diagnostics

Please try latest version and feel free to open a new issue if problem still exists.
